src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass.cs

```diff
@@ -87,13 +89,16 @@ public override BoundNode VisitTryStatement(BoundTryStatement node)
             return null;
         }
 
-        protected virtual void VisitTryBlock(BoundStatement tryBlock, BoundTryStatement node, ref TLocalState tryState)
+        protected virtual void VisitTryBlock(BoundBlock tryBlock, BoundTryStatement node, ref TLocalState tryState)
         {
+            var oldPending = SavePending();
             VisitStatement(tryBlock);
+            RestorePending(oldPending);
```

Comment:
Won't this remove `yield return` branches added in the `try`? Don't we want those?

Reply:
Restore removes labels and branches resolved to those labels. Other pending branches are carried forward. Yield is ok  (it does not actually have a label)

---
In reply to: [151245069](https://github.com/dotnet/roslyn/pull/23172#discussion_r151245069) [](ancestors = 151245069)